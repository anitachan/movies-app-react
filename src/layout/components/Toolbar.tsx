import { AppBar, Box, IconButton, Toolbar as MuiToolbar, Typography } from '@mui/material';
import MenuIcon from '@mui/icons-material/Menu';
import LanguageIcon from '@mui/icons-material/Language';
import SearchIcon from '@mui/icons-material/Search';
import { ThemeToggleButton } from './ThemeToggleButton';

interface Props {
	open: boolean;
	handleDrawerOpen: () => void;
	drawerWidth: number;
}

export const Toolbar = ({ open, handleDrawerOpen, drawerWidth }: Props) => {
	return (
		<AppBar
			position="fixed"
			sx={{
				width: open ? `calc(100% - ${drawerWidth}px)` : '100%',
				ml: open ? `${drawerWidth}px` : 0,
				transition: 'width 0.3s ease, margin 0.3s ease',
			}}
		>
			<MuiToolbar>
				{!open && (
					<IconButton color="inherit" aria-label="open drawer" onClick={handleDrawerOpen} edge="start" sx={{ mr: 2 }}>
						<MenuIcon />
					</IconButton>
				)}
				<Typography variant="h6" noWrap component="div" sx={{ flexGrow: 1, textAlign: 'center' }}>
					Movies
				</Typography>
				<Box sx={{ display: 'flex', gap: 1 }}>
					<IconButton aria-label="language" color="inherit">
						<LanguageIcon />
					</IconButton>
					<IconButton aria-label="search" color="inherit">
						<SearchIcon />
					</IconButton>
					<ThemeToggleButton />
				</Box>
			</MuiToolbar>
		</AppBar>
	);
};
