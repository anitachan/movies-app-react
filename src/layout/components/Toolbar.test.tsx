import { render, screen } from '@testing-library/react';
import { Toolbar } from './Toolbar';

const mockLocation = {
  pathname: '/foo/bar',
  search: '',
  hash: '',
  state: null,
  key: 'test',
};

jest.mock(
  'react-router-dom',
  () => ({
    useLocation: () => mockLocation,
  }),
  { virtual: true },
);

describe('Toolbar', () => {
  test('should render toolbar', () => {
    const { container } = render(
      <Toolbar open={true} drawerWidth={100} handleDrawerOpen={jest.fn()} />,
    );

    expect(container).toMatchSnapshot();
  });

  test('should show menu icon when open is false', () => {
    const { container } = render(
      <Toolbar open={false} drawerWidth={100} handleDrawerOpen={jest.fn()} />,
    );

    const menuIcon = screen.getByTestId('menu-icon');

    expect(menuIcon).toBeInTheDocument();
  });
});
