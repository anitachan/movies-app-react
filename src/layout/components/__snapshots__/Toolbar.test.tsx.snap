// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar should render toolbar 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1yuujcd-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ygil4i-MuiToolbar-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-17byzhr-MuiTypography-root"
      >
        <nav
          aria-label="breadcrumb"
          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1d5226r-MuiTypography-root-MuiBreadcrumbs-root"
        >
          <ol
            class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <div
                class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-5lnhk4-MuiTypography-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-6b04s2-MuiTypography-root-MuiLink-root"
                >
                  Movies
                </a>
              </div>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                data-testid="NavigateNextIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-17ymgc2-MuiTypography-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                  data-testid="HomeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                  />
                </svg>
                foo
              </p>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                data-testid="NavigateNextIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-17ymgc2-MuiTypography-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                  data-testid="HomeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                  />
                </svg>
                bar
              </p>
            </li>
          </ol>
        </nav>
      </div>
      <div
        class="MuiBox-root css-1i27l4i"
      >
        <button
          aria-label="language"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-h7cjts-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="LanguageIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2m6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56M12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96M4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2s.06 1.34.14 2zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56m2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8M12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96M14.34 14H9.66c-.09-.66-.16-1.32-.16-2s.07-1.35.16-2h4.68c.09.65.16 1.32.16 2s-.07 1.34-.16 2m.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56M16.36 14c.08-.66.14-1.32.14-2s-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2z"
            />
          </svg>
        </button>
        <button
          aria-label="search"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-h7cjts-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
            />
          </svg>
        </button>
        <div>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mcxdug-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="brightness4"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;
